libs/myReact/useState.ts

```diff
@@ -0,0 +1,29 @@
+let currentStateIndex = 0
+const states: any[] = []
+
+export default function useState<T>(
+	initialState: T,
+): [T, (newState: T | ((prevState: T) => T)) => void] {
```

Comment:
자바스크립트에서는 매개변수에 함수를 넘겨줄 때 콜백함수라고 부르지만, 리액트에서는 setState에 넘겨주는 함수를 updater function이라고 얘기를 해서 요렇게 표현해주면 의도가 더 명확할 것 같다는 생각도 드네요!
```suggestion
): [T, (newState: T | UpdaterFunction] {
```